dali/base/dadfs.cpp

```diff
@@ -3043,7 +3043,23 @@ class CDistributedFileBase : public CInterface, implements INTERFACE
     {
         return !checkProtectAttr(logicalName.get(),root,reason);
     }
-
+    bool checkOwned(StringBuffer &error)
+    {
+        Owned<IPropertyTreeIterator> iter = root->getElements("SuperOwner");
+        if (iter->first())
+        {
+            error.append("Cannot remove file ").append(logicalName.get()).append(" as owned by SuperFile(s): ");
+            loop
+            {
+                error.append(iter->query().queryProp("@name"));
+                if (!iter->next())
+                    break;
+                error.append(", ");
+            }
+            return true;
+        }
+        return false;
+    }
     bool canRemove(StringBuffer &reason,bool ignoresub=false)
```

Comment:
A light comment: I think the name 'reason' of the StringBuffer parameter is a little bit misleading. I can read it 'the reason of can remove'. Maybe the 'result' or something similar would be better.